src/main/java/tools/jackson/databind/ObjectMapper.java

```diff
@@ -563,6 +564,31 @@ public Collection<JacksonModule> getRegisteredModules() {
         return _savedBuilderState.modules();
     }
 
+    /*
+    /**********************************************************************
+    /* Collectors for Stream support.
+    /**********************************************************************
+    */
+
+    /**
+     * Creates a {@link Collector} that collects {@link JsonNode} elements into an {@link ArrayNode}.
+     * <p>
+     * This method uses this instance of {@link ObjectMapper} to create an empty {@link ArrayNode} and then adds each
+     * {@link JsonNode} to it.
+     * </p>
+     *
+     * @return a {@link Collector} that collects {@link JsonNode} elements into an {@link ArrayNode}
+     *
+     * @since 3.0
+     */
+    public Collector<JsonNode, ArrayNode, ArrayNode> toJsonNode() {
```

Comment:
Collector might be an overkill? How about simple `ObjectNode.asCollection()` or something?

Reply:
Sorry, I didn't fully understand your question.

The reason a Collector is returned is because `java.util.stream.Stream.collect()` requires a `java.util.stream.Collector`.

**Params**: `collector` – the Collector describing the reduction
**Type parameters**: 
`<R>` – the type of the result 
`<A>` – the intermediate accumulation type of the Collector

The proposal is to provide additional support for Java's Stream API by implementing a collector to JsonNode.

The reason I didn't propose a static call such as `ObjectNode.asCollection()` was due to @cowtowncoder suggestion to use existing main API types.

Could you expand/clarify a little bit more on your question? Thank you.

Reply:
Since result is a `Collector` should method name reflect that?

`jsonNodeCollector(...)` or something similar? `toJsonNode()` would indicate return value is `JsonNode` which it isn't.

Reply:
> Since result is a `Collector` should method name reflect that?
> 
> `jsonNodeCollector(...)` or something similar? `toJsonNode()` would indicate return value is `JsonNode` which it isn't.

Consider the `java.util.stream.Collectors` for instance - which was what I tried to mimic.

You can perform ```Collector.toList()```, ```Collector.toSet```, ```Collector.toMap```, etc...

None of these calls return a List, a Set or a Map. They return a Collector.of(). The methods reference the data structure that will collect each object in the stream.

`toList()` will collect `T` into a `List<T>`, this is the return value.

That's why I named it toJsonNode, because our returned data-structure is a JsonNode<JsonNode>.

### From java docs
>Returns a Collector that accumulates the input elements into a new Collection, in encounter order. The Collection is created by the provided factory.

**Reference**: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html

I hope I helped clarify, but if I didn't I'm happy to expand more on it.

Reply:
Right, but `ObjectMapper` != `Collectors`. So to me, name ought to indicate it relates to Collectors. 

So...

* `collector()`
* `nodeCollector()`
* `collectorForNodes()`

or something along those lines.

Alternatively perhaps helper object under `...util` package --`JacksonCollectors`?  -- wouldn't be a bad idea. Unless... I think I am missing somethign obvious.

Reply:
> Right, but `ObjectMapper` != `Collectors`. So to me, name ought to indicate it relates to Collectors.
> 
> So...
> 
> * `collector()`
> * `nodeCollector()`
> * `collectorForNodes()`
> 
> or something along those lines.
> 
> Alternatively perhaps helper object under `...util` package --`JacksonCollectors`? -- wouldn't be a bad idea. Unless... I think I am missing somethign obvious.

I agree.

My original idea was JacksonCollectors, because if we find more use cases for different types of collectors within databind in the future, then we will already have the right place for them.

Eitherway both are reasonable ideas.

Reply:
Oh. One small convenience thing is that to create `ArrayNode`, one needs `JsonNodeFactory` (or `ObjectMapper` / `ObjectReader` / `ObjectWriter` to act as one).
This is something `ObjectMapper` provides, over separate Collectors helper.

Reply:
Moved implementation as suggested.